apps/emqx_retainer/src/emqx_retainer_mnesia.erl

```diff
@@ -381,12 +381,24 @@ search_stream(Tokens, Now) ->
 
 search_stream(undefined, Tokens, Now) ->
     Ms = make_message_match_spec(Tokens, Now),
-    emqx_utils_stream:ets(
+    MsgStream = emqx_utils_stream:ets(
         fun
             (undefined) -> ets:select(?TAB_MESSAGE, Ms, 1);
             (Cont) -> ets:select(Cont)
         end
-    );
+    ),
+    case Tokens of
```

Comment:
Maybe a corresponding update is needed for the fdb retainer

Reply:
Will take care of that once the next sync PR is merged.